### PR TITLE
Feat/upgrade 3id did resolver

### PR DESCRIPTION
**Pull request by oed**
This PR upgrades the did-resolver package to v3 which has a new interface.

**Upgrades**
* `did-resolver`
* `did-jwt`
* `dids`

**Version bumps**
This PR also bumps the version of `key-did-resovler` and `@ceramicnetwork/3id-did-resolver` to v1 since the did-resolver has a breaking change to the resolver api.

### CODE REVIEW

**Comment by oed** [comment]
Need to look into why tests are breaking.

File: packages/3id-did-resolver/src/index.ts [hunk 1]
```diff
@@ -107,12 +102,31 @@ const resolve = async (ceramic: Ceramic, didId: string, commit?: string, v03ID?:
...
+      const response: DIDResolutionResult = {
+        didResolutionMetadata: { contentType },
+        didDocument: null,
+        didDocumentMetadata: {}
+      }
+      try {
         const version = getVersion(parsed.query)
-        return isLegacyDid(parsed.id) ? legacyResolve(ceramic, parsed.id, version) : resolve(ceramic, parsed.id, version)
+        const doc = await (isLegacyDid(parsed.id) ? legacyResolve(ceramic, parsed.id, version) : resolve(ceramic, parsed.id, version))
+        if (contentType === DID_LD_JSON) {
```
Comment (stbrody) [inline packages/3id-did-resolver/src/index.ts:116]: Can you add some comments on what these different cases represent?

**Review by stbrody — APPROVED** [review]
LGTM since I'm not really familiar enough with this stuff to give meaningful feedback.  The one potential suggestion I'd give would be to do a pass over the key-did-resolver and 3id-did-resolver to add more comments throughout so that someone other than you who isn't familiar with what it's doing can follow along.

File: packages/3id-did-resolver/src/index.ts (hunk 1 above)
Comment (oed) [reply packages/3id-did-resolver/src/index.ts:116]: The contentType?

File: packages/3id-did-resolver/src/index.ts [hunk 2]
```diff
@@ -10,57 +19,43 @@ interface Ceramic {
...
-    keyAgreement: []
+    keyAgreement: [],
+    publicKey: []
   }
-  const doc = Object.entries(content.publicKeys as string[]).reduce((diddoc, [keyName, keyValue]) => {
+  return Object.entries(content.publicKeys as string[]).reduce((diddoc, [keyName, keyValue]) => {
     const keyBuf = u8a.fromString(keyValue.slice(1), 'base58btc')
-    // remove multicodec varint
-    const publicKeyBase58 =  u8a.toString(keyBuf.slice(2), 'base58btc')
+    const entry: VerificationMethod = {
```
Comment (ukstv) [inline packages/3id-did-resolver/src/index.ts:34]: I am not sure if _all_ of the keys should get here. x25519 is meaningless for verification.

Does not hurt AFAIK, but really I am not sure if it should get here.

File: packages/3id-did-resolver/src/index.ts [hunk 3]
```diff
@@ -10,57 +19,43 @@ interface Ceramic {
...
-      })
-      // new keyAgreement format for x25519 keys
-      diddoc.keyAgreement.push({
-        id: `${did}#${keyName}`,
-        type: 'X25519KeyAgreementKey2019',
-        controller: did,
-        publicKeyBase58
-      })
+      entry.type = 'X25519KeyAgreementKey2019'
+      diddoc.verificationMethod.push(entry)
```
Comment (ukstv) [inline packages/3id-did-resolver/src/index.ts:48]: I am not sure if all of the keys should get here. x25519 is meaningless for verification.

Does not hurt AFAIK, but really I am not sure if it should get here.

Comment (oed) [reply packages/3id-did-resolver/src/index.ts:48]: It should be here according to the did-core spec.

Comment (ukstv) [reply packages/3id-did-resolver/src/index.ts:48]: ¯\_(ツ)_/¯

**Review by ukstv — APPROVED** [review]

**Pull request merged**